### PR TITLE
modbusdriver: retrieve exception information

### PR DESCRIPTION
**Pull request by Emantor**
We got a report of a user trying to access a modbus device with the
following error:
```
  Traceback (most recent call last):
  File "/opt/venvs/labgrid/lib/python3.9/site-packages/labgrid/remote/client.py", line 1638, in main
    args.func(session)
  File "/opt/venvs/labgrid/lib/python3.9/site-packages/labgrid/remote/client.py", line 814, in digital_io
    drv.set(False)
  File "/opt/venvs/labgrid/lib/python3.9/site-packages/labgrid/binding.py", line 100, in wrapper
    return func(self, *_args, **_kwargs)
  File "/opt/venvs/labgrid/lib/python3.9/site-packages/labgrid/driver/modbusdriver.py", line 36, in set
    raise ExecutionError('Could not write coil (code={})'.format(error_code))
  labgrid.driver.exception.ExecutionError: Could not write coil (code=7)
```
The error code is 7, indicating that an exception happened, however
labgrid did not retrieve the exception code, so no further debugging was
possible. Unify the modbusdriver error handling and retrieve the
exception if it happened. Also ignore the ACKNOWLEDGE and NONE exceptions.

Signed-off-by: Rouven Czerwinski <r.czerwinski@pengutronix.de>

**Description**
- [ ] CHANGES.rst has been updated

- [ ] PR has been tested

### CODE REVIEW

File: labgrid/driver/modbusdriver.py [hunk 1]
```diff
@@ -26,21 +26,31 @@ def on_activate(self):
     def on_deactivate(self):
         self.client = None
 
+    def handle_error(self, action):
+        error_code = self.client.last_error()
+        if error_code == 7:
+            exc = self.client.last_except()
+            if exc not in [0, 5]:
```
Comment (jluebbe) [inline labgrid/driver/modbusdriver.py:33]: Some comments for the numeric values would be useful. Perhaps they are already available as Enums in pyModbusTCP?

Comment (Emantor) [reply labgrid/driver/modbusdriver.py]: Done, We now additionally import the constants from pyModbusTCP and use instead of the magic values.

**Review by jluebbe — APPROVED** [review]
I've changed `handle_error` to a private function, as it should not be called from the outside.

**Comment by jluebbe** [comment]
@Emantor, please merge if you're fine with my change.

**Pull request merged**